src/include/OpenImageIO/imageio.h

```diff
@@ -2789,7 +2789,8 @@ inline bool attribute (string_view name, float val) {
 }
 /// Shortcut attribute() for setting a single string.
 inline bool attribute (string_view name, string_view val) {
-    const char *s = val.c_str();
+    std::string valstr = val;
+    const char *s = valstr.c_str();
```

Comment:
Why the temporary std::string here and not:
`const char *s = valstr.str().c_str();`
like you do throughout the rest of the PR?

Reply:
No reason. They mean the same thing. `valstr.str().c_str()` still makes a temporary string, it just doesn't have a name.

This entire set of changes is a actually from my vacation in December when I tinkered with a number of infrastructure oddities, just because (and when) I felt like it. I suspect that what happened here was that I worked on this a little bit at a time, fixed different places on different days and didn't check for absolutely consistency.

If you prefer the more compact one-expression idiom without the named temporary, I can change them all to be consistent.

Reply:
Actually, now that I look at it, C++17 std::string_view doesn't have the str() method either. So it turns out that this assignment to a std::string has the std::string_view equivalency I was aiming for, but the `sv.str().c_str()` idiom does not. The most compact 1-expression idiom that is fully interchangeable with std::string_view would be `std::string(sv).c_str()`.

I will fix this up to be std-equivalent everywhere.

Reply:
Notwithstanding the prior statement about how `sv.str().c_str()` should be the one that changes if we desire full interchangeability with std::string_view, now that I've had my coffee, I can see that there REALLY WAS a reason why we used the temporary in this place!

You need to look at the NEXT line, which is

    return attribute (name, TypeString, &s);

We need the *address* of the `char*`, which means the std::string that we called c_str() on needs to survive until after that call.

This particular spot was the correct idiom all along.

Reply:
@lgritz isn't this returning the address of a temporary in that case? Isn't that bad?

Reply:
Safety: It's taking the address of a named local variable, to pass to a function that will make a more permanent copy. So it's safe.

Copying: The old c_str() that this replaces made a ustring (!) to have something stable to take the c_str() of. Now it's just making a std::string, which is probably better performance, in general.

But most importantly, there is a second PR in flight, #3342, that makes an orthogonal set of changes that will eliminate the extra copies entirely.

Reply:
Thanks for the clarification and answering my silly questions @lgritz ;-). LGTM!